crates/runtime/src/accelerated_table/refresh_task_runner.rs

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2024 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use super::refresh_task::RefreshTask;
+use futures::future::BoxFuture;
+use tokio::{
+    select,
+    sync::mpsc::{self, Receiver, Sender},
+    task::JoinHandle,
+};
+
+use std::{
+    fmt::{Display, Formatter},
+    sync::Arc,
+};
+use tokio::sync::RwLock;
+
+use datafusion::{datasource::TableProvider, sql::TableReference};
+
+use super::refresh::Refresh;
+
+#[derive(Debug)]
+pub enum RefreshTaskResult {
+    Success,
+    Error,
+}
+
+impl Display for RefreshTaskResult {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            RefreshTaskResult::Success => write!(f, "Success"),
+            RefreshTaskResult::Error => write!(f, "Error"),
+        }
+    }
+}
+
+pub struct RefreshTaskRunner {
+    dataset_name: TableReference,
+    federated: Arc<dyn TableProvider>,
+    refresh: Arc<RwLock<Refresh>>,
+    accelerator: Arc<dyn TableProvider>,
+    task_thread: Option<JoinHandle<()>>,
+}
+
+impl RefreshTaskRunner {
+    #[must_use]
+    pub fn new(
+        dataset_name: TableReference,
+        federated: Arc<dyn TableProvider>,
+        refresh: Arc<RwLock<Refresh>>,
+        accelerator: Arc<dyn TableProvider>,
+    ) -> Self {
+        Self {
+            dataset_name,
+            federated,
+            refresh,
+            accelerator,
+            task_thread: None,
+        }
+    }
+
+    pub fn start(&mut self) -> (Sender<()>, Receiver<RefreshTaskResult>) {
+        let (task_sender, mut task_receiver) = mpsc::channel::<()>(1);
+
+        let (callback_sender, callback_receiver) = mpsc::channel::<RefreshTaskResult>(1);
```

Comment:
Lets rename these to be more descriptive:

`callback_sender` -> `notify_refresh_complete`
`callback_receiver` -> `on_refresh_complete`

Reply:
Updated - thank you